dashboard/app/models/stage.rb

```diff
@@ -152,4 +151,9 @@ def lockable_state(students)
       }
     end
   end
+
+  # Ensures we get the cached ScriptLevels, vs hitting the db.
+  def cached_script_levels
+    script_levels.map{|sl| Script.cache_find_script_level(sl.id)}
```

Comment:
Probably I'm missing something... But why access the cache to find the script_levels after iterating over them?